### PR TITLE
fix(widgets): wrong casting in some widgets when calling lv_event_get…

### LINKED ISSUE #5083: Wrong casting in some widgets when calling lv_event_get_param.
### Perform all steps below and tick them with [x]
- [ x] Read the [FAQ](https://docs.lvgl.io/master/intro/index.html#faq)
- [ x] Check the related part of the [Documentation](https://docs.lvgl.io/)
- [ x] Update lvgl to the latest version
- [- ] Reproduce the issue in a [Simulator](https://docs.lvgl.io/latest/en/html/get-started/pc-simulator.html)

### Describe the bug
Some widgets will not work properly on big endian architectures. I noticed it when using the tabview, which are using the button matrix, with a keypad. The tabs will then not change as expected (as it does in the simulator).

In case of events with code LV_EVENT_KEY, lv_event_get_param is returing an uint32_t pointer. However, in several widgets, this is casted to a char pointer:
`char c = *((char *)lv_event_get_param(e));`

In most cases this will work, but in big endian architectures this will always give c=0. I have found this issue in the following files, but I may have missed some:
lv_arc.c
lv_btnmatrix.c
lv_dropdown.c 
lv_obj.c
lv_roller.c
lv_slider.c

The solution should be to just replace the char with  uint32_t
`uint32_t c = *((uint32_t*)lv_event_get_param(e));`

### To Reproduce
Just add a tab view, or button matrix, to a big endian architecture.

### Expected behavior
The tabs in tabview should change also while using a keypad on a big endian architecture.

### Screenshots or video
NA

### PR DESCRIPTION
**Pull request by jimbom**
Help us review this PR! Anyone can [approve it or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### Description of the feature or fix
Fixes: #5083

All lines with `char c = *((char *)lv_event_get_param(e));` have been replaced with `uint32_t c = lv_event_get_key(e);`, as suggested in issue #5083.

### Checkpoints
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)

### CODE REVIEW

**Comment by kisvegabor** [comment]
Thank you! I've just fixed the failing tests.

**Review by kisvegabor — APPROVED** [review]
And CI is passed, merging.

Thank you! :blush:

**Comment by kisvegabor** [comment]
Oh, as I wasa the last committer, we need an approval from someone else too.

**Review by C47D — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Comment by FASTSHIFT** [comment]
Merged.

**Pull request merged**
